src/vs/workbench/browser/parts/editor/tabsTitleControl.ts

```diff
@@ -963,8 +966,9 @@ export class TabsTitleControl extends TitleControl {
 		tabContainer.title = title;
 
 		// Label
+		const displayName = editor.iconPath ? `$(${editor.iconPath.id}) ${escapeCodicons(name)}` : escapeCodicons(name);
```

Comment:
This logic is ugly and has to duplicated. We should look into pushing the icon logic into `ResourceLabel`